### ISSUE TITLE
[New build - DEFAULT] 20w21b: Turret Improvements.

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#2110]
Turret Computers were improved with a better handling of children entity and a preview while selected with the link tool.
Some refactorings were done regarding command enter/control/exit and player lock to entity: if you find anything suspicious regarding this please report it on the github! 

Changes and new features:
 - Character walk animation only if aligned to the ground.
 - Character Asteroid collision, still wip
 - Only one logical instruction carried per tick for each element.
 - Turret chain axis visualization. Green for a valid chain, red when an element is not part of a chain.
 - New turret contrain: the weapon must be on the same entity as the second axis
 - New turret contrain: the second axis must be a direct children of the first axis
 - Turret can be build on a children entity.
 
 Community Suggestion
 #2099 Hovercraft smooth power down.
 #2093 Separate Mouse sensitivity for vehicle control.

Bug Fixes:
#2102 Turret warning message when used with Camera.
#2103 Some Logic system could stall the game
#2098 Able to fly through the sea bed.
#2101 Torus Target widget still visible when close.
#2038 Multitool yellow box on startup.
#2048 Physics issue with Hovercraft with no repulsors.
#2105 Build on a fence is offset.
#2031 Seat exit move player too far.
#2058 Repulsor incorrect center of thrust issue.
#2104 Turret freeze while on turret is on children entity.
#2060 Turret on children entity not working.
#1637 Children Entity collision not updated on brick delete.

**Comment by tsunamayo** [comment]
Build is live! Thanks a lot! Still a lot of cases to fix... And you guys keep coming with new one ;)

**Comment by ZachZent** [comment]
![](https://cdn.discordapp.com/attachments/406664114249793536/712564132565745684/unknown.png)

**Comment by tsunamayo** [comment]
Lol I dont know if that was clear...

**Comment by ZachZent** [comment]
No it makes sense, It just reminded me of the three laws. Similar in tone while can't breaking the previous.

Glad to see asteroids are becoming solid, though janky as they currently are. Hope to place a starter block on one soon

**Comment by tsunamayo** [comment]
yeah asteroid is still wip, as they are moving it is actually hard to make it work correctly with the player... So actually it is more a player movement issue, I will need to make it more physicly accurate...

**Comment by ZachZent** [comment]
Yea that is understandable. I would slow down the asteroid spin a little bit as when you will be able to place a starter block to build on the asteroid, the amount of spinning would get a little nauseous

**Comment by tsunamayo** [comment]
so what I want is that once you start building / destroying it, it becomes an independent entity with its mass and velocity. So if you put some thruster on it it would stabilize and cancel the spin.

**Comment by ZachZent** [comment]
That sounds great! Still would suggest slowing the spin for the asteroids, just a little. Or better yet smaller asteroids initially spin faster while larger ones spin slower. Or just randomize it, that works too.
